app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -209,33 +223,42 @@ class UndertakingController @Inject()(
 
   def postCheckAnswers: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
+
     cyaForm.bindFromRequest().fold(
-      _ => throw new IllegalStateException("value hard-coded, form hacking?"),
+      _ => sys.error("value hard-coded, form hacking?"),
       form => {
-        store.update[UndertakingJourney]({ x =>
-          x.map { y =>
-            y.copy(cya = y.cya.copy(value = Some(form.value.toBoolean)))
-          }
-        }).flatMap { journey: UndertakingJourney =>
-          for {
-            ref <- escService.createUndertaking(
-                    Undertaking(
-                      None,
-                      name = UndertakingName(journey.name.value.getOrElse(throw new IllegalThreadStateException(""))),
-                      industrySector = journey.sector.value.getOrElse(throw new IllegalThreadStateException("")),
-                      None,
-                      None,
-                      List(BusinessEntity(eori, leadEORI = true, journey.contact.value)
-                    )))
-
-          } yield {
-            Redirect(routes.UndertakingController.getConfirmation(ref, journey.name.value.getOrElse("")))
+        store.update[UndertakingJourney]({ _.map { undertakingJourney =>
+          val updatedCYA = undertakingJourney.cya.copy(value = form.value.toBoolean.some)
+          undertakingJourney.copy(cya = updatedCYA)
           }
+        }).flatMap { updatedUndertakingJourney: UndertakingJourney =>
+          val undertaking = Undertaking(
+            None,
+            name = UndertakingName(updatedUndertakingJourney.name.value.getOrElse(throw new IllegalThreadStateException(""))),
```

Comment:
Is an `IllegalThreadStateException` the right exception to throw here? Also setting a message will be helpful.

Reply:
It's part of old code and at lot of places exceptions are being thrown which don't fit :) I tried to change some but some are pending. will try to change that

Reply:
Ok, definitely worth looking at alternatives to throwing exceptions too since there's usually a better way.